src/bin/console/console.cpp

```diff
@@ -351,14 +356,26 @@ void Console::out(const std::shared_ptr<const Table>& table, uint32_t flags) {
 int Console::_exit(const std::string&) { return Console::ReturnCode::Quit; }
 
 int Console::_help(const std::string&) {
+  auto encoding_options = std::string{"                                               Encoding options: "};
+  encoding_options += boost::algorithm::join(
+      encoding_type_to_string.right | boost::adaptors::transformed([](auto it) { return it.first; }), ", ");
+  // Split the encoding options in lines of 120 and add padding
+  auto line_wrap = std::regex{"(.{1,120})(?: +|$)"};
```

Comment:
I can not read this. Can a stupid bloody function with a loop do this? What the hell is "?: +|$"? "-50", where does that number come from?

Reply:
\*sobs profoundly*

Reply:
I'll grudgingly replace it, but tbh, I think that regular expressions are something that programmers should know, love, and use. It's not like this is an [e-mail validation regex](https://emailregex.com/) or uses non-regular stuff like lookarounds or non-greedy behavior.

Reply:
After looking into the options with a loop, I found that the stackoverflow answers mistreat new lines in the input data. Fixing this and adding support for padding would blow this up. I strongly consider the regex version to be better, but agree to add an explanatory comment.

Reply:
Is making this a nice little utility function harder than this:

```
words = boost::split_this_by_newline_or_whitespace(str)
lines = [str()]
while !words.empty():
  if !lines.back().empty() or len(lines.back() + words.front()) > 120:
    lines.append(str())
  lines.back.append(words.pop_front())

return boost::join_this_stuff_with_newlines(lines)
```
?

But in the end, of course, it's up to you.

Reply:
If I am not mistaken, implementation would suffer from the same problems as the Stack Overflow solutions (mistreating multi-line input). To solve that, you would first have to split the input by lines, then by words.